libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/pact/ContractTest.java

```diff
@@ -0,0 +1,751 @@
+                .given("expiry is 2099-01-01 00:00:00")
+                .given("a contra-indicator is returned with code TEST-CI-CODE-2 for a passport")
+                .uponReceiving("Request for contra-indicators for user with no contra-indicators.")
+                .path(GET_VCS_ENDPOINT)
+                .query("user_id=" + MOCK_USER_ID)
+                .method("GET")
+                .headers(
+                        IP_ADDRESS_HEADER,
+                        MOCK_IP_ADDRESS,
+                        GOVUK_SIGNIN_JOURNEY_ID_HEADER,
+                        MOCK_GOVUK_SIGNIN_ID,
+                        PactDslRequestBase.CONTENT_TYPE,
+                        "application/json")
+                .willRespondWith()
+                .status(200)
+                .body(responseForGetCi)
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "getCisUserIdReturnsNoCisVc")
+    void fetchContraIndicators_whenCalledWithUserIdAgainstCimiApi_receivesEmptyContraIndicators(
+            MockServer mockServer) throws CiRetrievalException {
+        // Arrange
+        when(mockConfigService.getParameter(CIMIT_COMPONENT_ID)).thenReturn(TEST_ISSUER);
+        when(mockConfigService.getParameter(CIMIT_SIGNING_KEY)).thenReturn(EC_PRIVATE_KEY_JWK);
+        when(mockConfigService.getParameter(CIMIT_API_BASE_URL))
+                .thenReturn(getMockApiBaseUrl(mockServer));
+        var underTest = new CiMitService(mockConfigService);
+
+        // Act
+        var contraIndicator =
+                underTest.getContraIndicatorsVc(
+                        MOCK_USER_ID, MOCK_GOVUK_SIGNIN_ID, MOCK_IP_ADDRESS);
+
+        // Assert
+        assertEquals(MOCK_USER_ID, contraIndicator.getUserId());
+        assertInstanceOf(SecurityCheckCredential.class, contraIndicator.getCredential());
+
+        var securityCheckCredential = (SecurityCheckCredential) contraIndicator.getCredential();
+        var evidence = (SecurityCheck) securityCheckCredential.getEvidence().get(0);
+
+        assertEquals(0, evidence.getContraIndicator().size());
+    }
+
+    @Pact(provider = "CiMitProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact postCiSuccessfullyPostsContraIndicator(PactDslWithProvider builder) {
+        var responseForPostCi = newJsonBody(body -> body.stringValue("result", "success")).build();
+
+        return builder.given("mockUserId is the user")
+                .given("mockCimitComponentId is the issuer")
+                .given("the current time is 2024-01-01 00:00:00")
+                .given("the VC is from DCMAW-5477-AC1")
+                .given("the VC has CI code TEST-CI-CODE")
+                .uponReceiving(
+                        "Request for contra-indicators for specific user with existing contra-indicators.")
+                .path(POST_CI_ENDPOINT)
+                .method("POST")
+                .body(String.format("{\"signed_jwt\": \"%s\"}", FAILED_DVLA_VC_WITH_CI_JWT))
```

Comment:
Not an issue, but could imagine using the Jackson 
```
objectMapper.writeValueAsString(Map.of("signed_jwt", FAILED_DVLA_VC_WITH_CI_JWT))
```
could have worked for serialising this object